src/autorest-core/lib/configuration.ts

```diff
@@ -309,18 +310,32 @@ export class ConfigurationView {
       // update source locations to point to loaded Swagger
       if (m.Source) {
         const blameSources = m.Source.map(s => {
-          try {
-            const blameTree = this.DataStore.Blame(s.document, s.Position);
-            const result = [...blameTree.BlameInputs()];
-            if (result.length > 0) {
-              return result.map(r => <SourceLocation>{ document: r.source, Position: Object.assign(TryDecodeEnhancedPositionFromName(r.name) || {}, { line: r.line, column: r.column }) });
+          let posClone = Clone(s.Position);
+          while (true) {
```

Comment:
control flow is super hard to analyze (at least for me) due to this loop and the numerous breaks and the return spilled across it! It does look sane to me, but I'd rather have this loop tighter and more expressive!

How about something like:
``` typescript
let blameTree: BlameTree | null = null;
while (blameTree === null) {
  try {
    blameTree = ...
  } catch (e) {
    // ~ current catch logic, message should also include `e` somehow
  }
}
const result = [...blameTree.BlameInputs()];
... // old logic
```